pkg/controller/installation/core_controller.go

```diff
@@ -877,9 +898,9 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 			log.Info("Rebooting to enable TigeraSecure controllers")
 			os.Exit(0)
 		} else if err != nil {
-			r.SetDegraded("Error discovering Tigera Secure availability", err, reqLogger)
+			r.SetDegraded(operator.ResourceNotFound, "Error discovering Tigera Secure availability", err, reqLogger)
```

Comment:
This should not be a ResourceNotFound error.

I am not entirely sure what it should be. Maybe we can have an InternalServerError or something like that to catch unexpected errors, similar to HTTP 500?

Reply:
There is another similar "Error discovering Tigera Secure availability" as well below.